servers/rendering/renderer_rd/shaders/canvas.glsl

```diff
@@ -48,6 +48,8 @@ layout(set = 1, binding = 0, std140) uniform MaterialUniforms {
 /* clang-format on */
 #endif
 
+InstanceData draw_data;
```

Comment:
As we discussed in the meeting, only `instance_index` should be in global scope since anything in global scope will occupy a register for the whole shader invocation. We definitely don't want the entire InstanceData struct in registers for the whole shader